### PR TITLE
Allow Spree::Admin::UserPasswordsController to be accessed from admin…

### PR DESCRIPTION
**Pull request by acreilly**
… panel.

Since the Spree::Admin::UserPasswordsController was not being used anywhere within solidus the routes were not working. In order to get it working I needed to add a custom route in routes.rb because the default one was being overridden by the frontend route. Since we are calling this endpoint from the admin, we need to skip the action require_no_authentication because we are already authorizing the admin elsewhere. Without it we receive the error "Filter chain halted as :require_no_authentication rendered or redirected"

### CODE REVIEW

File: lib/controllers/backend/spree/admin/user_passwords_controller.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class Spree::Admin::UserPasswordsController < Devise::PasswordsController
   helper 'spree/admin/tables'
   layout 'spree/layouts/admin'
 
+  skip_before_action :require_no_authentication, only: [:create]
```
Comment (mamhoff) [inline lib/controllers/backend/spree/admin/user_passwords_controller.rb]: Can we require authentication here? We don't want this route to be open to non-admins.

Comment (ericsaupe) [inline lib/controllers/backend/spree/admin/user_passwords_controller.rb]: The route is to send the reset email which is open to people without authentication. For example, you aren't logged in when you request a recovery email. Skipping this action allows logged in admins to request the recovery email.

**Comment by acreilly** [comment]
This PR depends on #96 . Once that is merged, I rebase and utilize that PRs routes and add a comment next to the skip_before_action : require_no_authentication explaining how it is needed since the admin is logged in while hitting the create endpoint.

File: lib/controllers/backend/spree/admin/user_passwords_controller.rb (hunk 1 above)
Comment (tvdeyen) [reply lib/controllers/backend/spree/admin/user_passwords_controller.rb]: 👍 The `require_no_authentication` does not open this url to non-admin users, it skips the default behaviour of Devise that only allows to create a password reset request for non-logged in users, what is not practical for admins. They would need to log out to send a password reset token. The actual password reset is handled another controller (`lib/controllers/frontend/spree/user_passwords_controller.rb`)

**Comment by tvdeyen** [comment]
@acreilly now that #96 gets merged would you mind to rebase with current `master`?

**Comment by aitbw** [comment]
Hey @acreilly, thanks a lot for this PR! Would you mind rebasing it against `master`?

**Comment by acreilly** [comment]
Closing and opened https://github.com/solidusio/solidus_auth_devise/pull/136

**Pull request closed without merging**